# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My provisional lane is **Refresh / Content Opportunity Scoring**.

I chose this lane because organizations often manage thousands of content pages, making it impossible to review every page manually. The goal is to support content or SEO specialists in deciding which pages should be reviewed first by ranking them according to their potential improvement opportunities. Rather than simply training a machine learning model, this project focuses on improving a real decision-making process through data-driven prioritization. This choice is provisional and may evolve as I gain a better understanding of the data during the first weeks of the project.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The decision this project aims to improve is **which content pages should be reviewed first**.

The primary users are content editors or SEO specialists. They would use the ranked recommendations to decide which pages should be refreshed, expanded, monitored, or investigated further.

The output of the project will be a ranked list of content pages, together with a priority score and supporting reason codes explaining why each page was recommended.

A false positive would recommend a page that does not actually require attention, wasting valuable review time and resources. A false negative would fail to identify a page that could have benefited from review, causing a potentially valuable opportunity for improvement to be missed.

Machine learning is appropriate because page performance depends on many interacting signals, such as traffic, engagement, content characteristics, and search performance. These relationships are too complex to capture reliably with a single hand-written rule.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Clients:", df["client_id"].nunique())

df["is_declining_label"] = (
    df["trend_direction"].astype(str).str.lower() == "down"
).astype(int)

print("\nDeclining label counts:")
print(df["is_declining_label"].value_counts())

print("\nDeclining label percentages:")
print(df["is_declining_label"].value_counts(normalize=True) * 100)



Rows: 30000
Columns: 44
Clients: 32

Declining label counts:
is_declining_label
1    16262
0    13738
Name: count, dtype: int64

Declining label percentages:
is_declining_label
1    54.206667
0    45.793333
Name: proportion, dtype: float64


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

The starter dataset contains **30,000 content items** from **32 clients**, making manual review of every page unrealistic. This supports the idea of building a system that helps prioritize which pages should be reviewed first.

After creating the starter label, **54.2%** of the pages are labelled as declining, while **45.8%** are not. This suggests that the dataset contains enough examples of both outcomes to investigate content prioritization methods.

These observations do not prove that a machine learning model will solve the problem, but they suggest that the dataset is large enough and sufficiently diverse to explore this research question.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.